ThePortugueseMan.HabitTracker/Habit_DataBase/DataBaseCommands.cs

```diff
@@ -0,0 +1,394 @@
+            }
+            Console.WriteLine("\n-----------------------------");
+        }
+    }
+    //returnType == "TableName" returns the name of the table at index of the mainTable
+    //returnType == "HabitUnit" return the name of the habit's unit ^""
+    public string? GetTableNameOrUnitsFromIndex(string? mainTableName, int index, string? returnType)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            string? returnString = null;
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                $"SELECT * FROM {mainTableName} WHERE Id = {index}";
+
+            SqliteDataReader reader = tableCmd.ExecuteReader();
+
+            if (!reader.HasRows)
+            {
+                connection.Close();
+                return null;
+            }
+            else
+            {
+                reader.Read();
+                if (returnType == "TableName") returnString = reader.GetString(1);
+                else if (returnType == "HabitUnit") returnString = reader.GetString(2);
+
+                reader.Close();
+                return returnString;
+            }
+        }
+    }
+    public string? GetUnitFromTableName(string? mainTableName, string? subTableName)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            string? returnString = null;
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                $"SELECT HabitUnit FROM {mainTableName} WHERE HabitTableName = '{subTableName}'";
+
+            SqliteDataReader reader = tableCmd.ExecuteReader();
+
+
+            if (!reader.HasRows)
+            {
+                connection.Close();
+                return null;
+            }
+            else
+            {
+                reader.Read();
+                returnString = reader.GetString(0);
+                reader.Close();
+                return returnString;
+            }
+        }
+    }
+    public bool ChangeSubTableName(string? currentTableName, string? newTableName)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            try
+            {
+                tableCmd.CommandText =
+                $"ALTER TABLE {currentTableName} RENAME TO {newTableName}";
+
+                tableCmd.ExecuteNonQuery();
+                connection.Close();
+                return true;
+            }
+            catch { return false; }
+        }
+    }
+    public class SubTable
+    {
+        public int Id { get; set; }
+        public DateTime Date { get; set; }
+        public int Quantity { get; set; }
+    }
+    //represents an entry in the main table - a habit
```

Comment:
Please check the necessity of this comment. 😊 The code seems self-explanatory to me.